shared/fetcherAnonymous.ts

```diff
@@ -0,0 +1,65 @@
+import axios from "axios";
+import Cookies from "js-cookie";
+
+import getConfig from "next/config";
+import { ITokenEnum } from "@shared/enum";
+
+const { publicRuntimeConfig } = getConfig();
+
+const baseURL = publicRuntimeConfig.BASE_URL;
+
+const api = axios.create({
+  baseURL,
+  timeout: 10000,
+  headers: { "content-type": "application/json"},
+});
+api.interceptors.request.use(
+  async (config) => {
+    const token = Cookies.get(ITokenEnum.anonymousToken);
+    if (token) {
+      config.headers.Authorization = `Bearer ${token}`;
+    }
+    return config;
+  },
+  (error) => Promise.reject(error)
+);
+
+api.interceptors.response.use(
+  (response) => response,
+  async (error) => {
+    if (
+      error?.response?.status === 401 &&
+      error?.response?.config?.url !== "/users/refresh-token"
+    ) {
+      const token = Cookies.get(ITokenEnum.refreshToken);
+      const refreshApi = axios.create({
+        baseURL,
+        timeout: 3000,
+        headers: {
+          "content-type": "application/json",
+          authorization: `Bearer ${token}`,
+        },
+      });
+      return refreshApi.get("/users/refresh-token").then(
+        async (refreshResponse) => {
+          const data = refreshResponse?.data?.data;
+          if (data?.token) {
+            Cookies.set(ITokenEnum.token, data?.token);
+            Cookies.set(ITokenEnum.refreshToken, data?.refresh_token);
+            Cookies.set(ITokenEnum.anonymousToken, data?.anonymousToken);
+            error.config.headers.Authorization = `Bearer ${data?.token}`;
+            return axios.request(error?.config);
+          }
+          return Promise.reject(error);
+        },
+        (refreshError) => {
+          console.log("[refreshError]: ", refreshError);
+          return Promise.reject(error);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Avoid using `console.log` for error logging in production code. It's better to use a dedicated logging service that provides more control over log levels, formats, and destinations. If you don't have a logging service, at least consider replacing `console.log` with `console.error`.
```diff
- console.log("[refreshError]: ", refreshError);
+ console.error("[refreshError]: ", refreshError);
```

Overall, this code does a good job of setting up an Axios instance with interceptors for request and response. The interceptors add an authorization header to each request and handle token refresh on receiving a 401 status code. However, the error handling could be improved as suggested above.


<!-- This is an auto-generated comment by OSS CodeRabbit -->